tests/foreman/cli/test_flatpak.py

```diff
@@ -0,0 +1,182 @@
+
+    # 2. Ensure that remotes can be created only with proper permissions.
+    p = 'create_flatpak_remotes'
+    with pytest.raises(CLIReturnCodeError) as e:
+        target_sat.cli.FlatpakRemote().with_user(usr, pwd).create(
+            {
+                'organization-id': function_org.id,
+                'url': FLATPAK_REMOTES['Fedora']['url'],
+                'name': gen_string('alpha'),
+            }
+        )
+    assert emsg.format(p) in str(e)
+
+    target_sat.api_factory.create_role_permissions(function_role, {'Katello::FlatpakRemote': [p]})
+    remote = (
+        target_sat.cli.FlatpakRemote()
+        .with_user(usr, pwd)
+        .create(
+            {
+                'organization-id': function_org.id,
+                'url': FLATPAK_REMOTES['Fedora']['url'],
+                'name': gen_string('alpha'),
+            }
+        )
+    )
+    res = target_sat.cli.FlatpakRemote().with_user(usr, pwd).info({'name': remote['name']})
+    assert res == remote, 'Read values differ from the created ones'
+
+    # 3. Ensure that remotes can be updated and scanned only with proper permissions.
+    p = 'edit_flatpak_remotes'
+    desc = gen_string('alpha')
+    with pytest.raises(CLIReturnCodeError) as e:
+        target_sat.cli.FlatpakRemote().with_user(usr, pwd).update(
+            {'name': remote['name'], 'description': desc}
+        )
+    assert emsg.format(p) in str(e)
+    with pytest.raises(CLIReturnCodeError) as e:
+        target_sat.cli.FlatpakRemote().with_user(usr, pwd).scan({'name': remote['name']})
+    assert emsg.format(p) in str(e)
+
+    target_sat.api_factory.create_role_permissions(function_role, {'Katello::FlatpakRemote': [p]})
+    target_sat.cli.FlatpakRemote().with_user(usr, pwd).update(
+        {'name': remote['name'], 'description': desc}
+    )
+    target_sat.cli.FlatpakRemote().with_user(usr, pwd).scan({'name': remote['name']})
+    res = target_sat.cli.FlatpakRemote().with_user(usr, pwd).info({'name': remote['name']})
+    assert res['description'] == desc, 'Description was not updated'
+    assert 'http' in res['registry-url'], 'Scan of flatpak remote failed'
+
+    # 4. Ensure that remotes can be deleted only with proper permissions.
+    p = 'destroy_flatpak_remotes'
+    with pytest.raises(CLIReturnCodeError) as e:
+        target_sat.cli.FlatpakRemote().with_user(usr, pwd).delete({'name': remote['name']})
+    assert emsg.format(p) in str(e)
+
+    target_sat.api_factory.create_role_permissions(function_role, {'Katello::FlatpakRemote': [p]})
+    res = target_sat.cli.FlatpakRemote().with_user(usr, pwd).delete({'name': remote['name']})
+    assert 'Flatpak Remote deleted' in res
+    with pytest.raises(CLIReturnCodeError) as e:
+        target_sat.cli.FlatpakRemote().with_user(usr, pwd).info({'name': remote['name']})
+    assert 'Error: flatpak_remote not found' in str(e)
+
+
+def test_scan(target_sat, function_org, function_product):
```

Comment:
Non-blocking: what do you think about renaming this to `test_scan_flatpak_remote`? `test_scan` seems a bit too terse to me.

Reply:
You are right @synkd, I'll update it.

I usually start writing a test case with some short name like `test_dummie` and update the name specifically later. The update didn't happen this time.